### PR TITLE
Adding an implementation for the knapsack problem.

### PR DESCRIPTION
**Pull request by karpinski**
Adding the implementation and a respective README entry.

### CODE REVIEW

File: dp/knapsack.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+def get_maximum_value(items, capacity):
+    dp = [0] * (capacity + 1)
+    for item in items:
+        for current_weight in range(capacity + 1):
+            total_weight = current_weight + item.weight
+            if total_weight <= capacity:
+                dp[total_weight] = max(dp[total_weight],
+                                       dp[current_weight] + item.value)
+    maximum_value = 0
```
Comment (ankit167) [inline dp/knapsack.py]: we could simply return **max(dp)** here

Comment (karpinski) [inline dp/knapsack.py]: Fair point! Done.

**Comment by ankit167** [comment]
@karpinski get_maximum_value([Item(60, 10), Item(100, 20), Item(120, 30)],50)) gives output 300 instead of 220

**Comment by karpinski** [comment]
@ankit167 My bad, it should be fine now.

**Pull request merged**